Analytics/Classes/Internal/SEGFileStorage.m

```diff
@@ -0,0 +1,164 @@
+//
+//  SEGFileStorage.m
+//  Analytics
+//
+//  Copyright © 2016 Segment. All rights reserved.
+//
+
+#import "SEGUtils.h"
+#import "SEGFileStorage.h"
+
+@interface SEGFileStorage ()
+
+@property (nonatomic, strong, nonnull) NSURL *folderURL;
+
+@end
+
+@implementation SEGFileStorage
+
+- (instancetype)init {
+    return [self initWithFolder:[SEGFileStorage applicationSupportDirectoryURL] crypto:nil];
+}
+
+- (instancetype)initWithFolder:(NSURL *)folderURL crypto:(id<SEGCrypto>)crypto {
+    if (self = [super init]) {
+        _folderURL = folderURL;
+        _crypto = crypto;
+        [self createDirectoryAtURLIfNeeded:folderURL];
+        return self;
+    }
+    return nil;
+}
+
+- (void)removeKey:(NSString *)key {
+    NSURL *url = [self urlForKey:key];
+    NSError *error = nil;
+    if (![[NSFileManager defaultManager] removeItemAtURL:url error:&error]) {
+        SEGLog(@"Unable to remove key %@ - error removing file at path %@", key, url);
+    }
+}
+
+- (void)resetAll {
+    NSError *error = nil;
+    if (![[NSFileManager defaultManager] removeItemAtURL:self.folderURL error:&error]) {
+        SEGLog(@"ERROR: Unable to reset file storage. Path cannot be removed - %@", self.folderURL.path);
+    }
+    [self createDirectoryAtURLIfNeeded:self.folderURL];
+}
+
+- (void)setData:(NSData *)data forKey:(NSString *)key {
+    NSURL *url = [self urlForKey:key];
+    if (self.crypto) {
+        NSData *encryptedData = [self.crypto encrypt:data];
+        [encryptedData writeToURL:url atomically:YES];
+    } else {
+        [data writeToURL:url atomically:YES];
+    }
+    
+    NSError *error = nil;
+    if (![url setResourceValue:@YES
+                        forKey:NSURLIsExcludedFromBackupKey
```

Comment:
If we do that responsibility should still live inside the storage implementation, but we can just provide an API on top like
`setBackup:(BOOL)backup forKey:(NSString *)key`